### PR TITLE
feat: line area series theme

### PR DESCRIPTION
**Pull request by jung-han**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

```ts
interface DotTheme {
  color?: string;
  radius?: number;
  borderColor?: string;
  borderWidth?: number;
}

interface theme {
  series: {
    colors?: string[];
    dot?: DotTheme;
    select?: {
      dot?: DotTheme;
      areaOpacity?: number; // selected series area opacity, 🌝  only area chart
      restSeries?: {
        areaOpacity?: number; // unselected series area opacity, 🌝 only area chart
      }
    }
    hover?: {
      dot?: DotTheme;
    }
  }
};
```

**line**
image

**area**
image

**line area**
image

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/component/areaSeries.ts [hunk 1]
```diff
@@ -391,37 +377,68 @@ export default class AreaSeries extends Component {
...
+      ...m,
+      type: 'areaPoints',
+      lineWidth: 0,
+      color: 'rgba(0, 0, 0, 0)', // make area border transparent
+      fillColor: this.getAreaOpacity(m.name!, m.color!),
+    }));
+  }
+
+  renderCircleModel({
+    options,
```
Comment (jungeun-cho) [inline src/component/areaSeries.ts:401]: `showDot`만 넘겨줘도되지 않을까요?

File: src/component/lineSeries.ts [hunk 2]
```diff
@@ -334,6 +346,17 @@ export default class LineSeries extends Component {
...
 
+  private getSelectedSeriesWithTheme(models: CircleResponderModel[]) {
+    const { dot } = this.theme.select;
+
+    return models.map((model) => ({
+      ...model,
+      radius: dot?.radius,
+      color: dot?.color ?? model.color,
+      style: ['hover', { lineWidth: dot!.borderWidth, strokeStyle: dot!.borderColor }],
+    }));
```
Comment (jungeun-cho) [inline src/component/lineSeries.ts:357]: dot테마가 항상 존재한다고 하면 `dot`을 가져와 설정하는건 어때요?
```ts
    const { radius, color, borderWidth, borderColor } = this.theme.select.dot as DotTheme;

    return models.map((model) => ({
      ...model,
      radius,
      color: color ?? model.color,
      style: ['hover', { lineWidth: borderWidth, strokeStyle: borderColor }],
    }));
```

File: src/helpers/theme.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    case 'treemap':
+    case 'heatmap':
+      return {
+        startColor: defaultSeriesTheme.startColor,
+        endColor: defaultSeriesTheme.endColor,
+      };
+    case 'pie':
+      return {};
+    default:
+      return lineTypeSeriesTheme;
```
Comment (jungeun-cho) [inline src/helpers/theme.ts:108]: default가 lineTypeSeriesTheme인가요?_?

**Review by jungeun-cho — APPROVED** [review]
[10/19] 리뷰완료했습니다. 이견없어요~
테마가 다양하게 적용되네요 ㅎㅎㅎ 고생하셨어요.

File: src/component/areaSeries.ts [hunk 4]
```diff
@@ -391,37 +377,68 @@ export default class AreaSeries extends Component {
...
+  }: RenderOptions): { dotSeriesModel: CircleModel[]; responderModel: CircleModel[] } {
+    const dotSeriesModel = [] as CircleModel[];
+    const responderModel = [] as CircleModel[];
```
Comment (lja1018) [inline src/component/areaSeries.ts:404]: `const XXX: CircleModel[] = [];`가 아닌 이렇게 사용한 이유가 궁금합니다!

**Review by lja1018 — APPROVED** [review]
[10/21] 리뷰완료
고생하셨습니다!

File: src/helpers/theme.ts (hunk 3 above)
Comment (jung-han) [reply src/helpers/theme.ts:108]: {}로 수정했어요!

**Pull request merged**